lua/ii/disting.lua

```diff
@@ -118,6 +118,56 @@ do return
   , { name = 'all_notes_off'
     , cmd = 0x57
     }
+  -- dual mode algorithms
+  , { name = 'set_dual_parameter'
+    , cmd = 0x5D
+    , docs = 'set side/parameter to actual value'
+    , args = { { 'side_parameter', u8 }
+             , { 'value', s16 }
+             }
+    }
+  , { name = 'set_dual_scale_parameter'
+    , cmd = 0x5E
+    , docs = 'The 0-16384 value range scales to the actual param. value range'
+    , args = { { 'side_parameter', u8 }
+             , { 'value', u16 }
```

Comment:
query if we can use `s16V` type here instead? in general we try and avoid scripters having to think in arbitrary scale ranges (eg 0-16384), instead opting for a more human friendly `volts` concept which is at least in base10.

the range [0..16384] is represented by [0.0 .. 10.0] with the `s16V` type. the underlying message that is sent is identical, so it's just a matter of the representation in a crow script.

i assume the values on disting itself are represented differently than as a raw 14bit value range -- if there is a numeric readout of values, i guess it would be best to mimic those values so one's script can speak in the same units as the module itself.

sorry if this is hard to grok - i've always had a hard time describing the custom type support in the i2c description lang.

Reply:
butting in here to note that the `s16v` values are kind of useless with disting i2c. To control disting via i2c you either use `set_parameter` and specify the literal parameter value or use `set_scale_parameter` / `set_controller` and send 0-16384. 0.0-10.0 is the arbitrary scale in this case.

Reply:
ok i just looked a little at a video and maybe i get it now?

it seems like the 'parameter value' is not a unified range, but each different parameter has a different range (but the maximum range is 0 - 16384). if that's correct, then i guess there's nothing to do but force scripters to deal with the absolute range.

so correct me if i'm wrong: in *some* cases where the range is arbitrarily mapped to (0, 16384) it would be nice to use s16V, and even better in cases where the parameter is pitch, but some cases just use the numbers as indices or a smaller range, so you need to have precise integer values.

if that makes sense, then i agree s16V is not a good fit for an mutating param range like this. but please correct me if the above is not the issue.

Reply:
All disting parameters can be set in one of two ways, either you can specify the actual value that you want with no scaling or mapping whatsoever or you can send a 14-bit value and let the disting map it to the corresponding value in the set of valid values for that parameter. For example, on the Poly Wavetable synth algorithm in the Disting EX parameter 57 is a "Chorus mode" parameter and the possible values are 0, 1, or 2. So you can either use i2c command `0x46` to set parameter 57 to 0, 1 or 2... or you can use i2c command `0x47` to set parameter 57 via a 14-bit value that the disting will then map to mode 0, 1 or 2. In neither case is a range of 0.0..10.0 sensible. None of the i2c commands on the disting have any concept of "volts", so 10.0 will never be treated as a maximum unless it just so happens to be for that particular parameter. It's more like a 14-bit MIDI implementation. @scanner-darkly has posted the i2c specification here: https://github.com/scanner-darkly/teletype/wiki/DISTING-EX-I2C-SPECIFICATION